tools/codegen/src/generator/languages/rust/reader/implementation.rs

```diff
@@ -150,27 +150,25 @@ impl ImplReader for ast::DynVec {
                     return ve!(Self, TotalSizeNotMatch, molecule::NUMBER_SIZE * 2, slice_len);
                 }
                 let offset_first = molecule::unpack_number(&slice[molecule::NUMBER_SIZE..]) as usize;
-                if offset_first % 4 != 0 || offset_first < molecule::NUMBER_SIZE * 2 {
+                if offset_first % molecule::NUMBER_SIZE != 0 || offset_first < molecule::NUMBER_SIZE * 2 {
                     return ve!(Self, OffsetsNotMatch);
                 }
-                let item_count = offset_first / 4 - 1;
-                let header_size = molecule::NUMBER_SIZE * (item_count + 1);
-                if slice_len < header_size {
-                    return ve!(Self, HeaderIsBroken, header_size, slice_len);
+                if slice_len < offset_first {
+                    return ve!(Self, HeaderIsBroken, offset_first, slice_len);
                 }
-                let mut offsets: Vec<usize> = slice[molecule::NUMBER_SIZE..]
-                    .chunks(molecule::NUMBER_SIZE)
-                    .take(item_count)
+                let mut offsets: Vec<usize> = slice[molecule::NUMBER_SIZE..offset_first]
+                    .chunks_exact(molecule::NUMBER_SIZE)
                     .map(|x| molecule::unpack_number(x) as usize)
                     .collect();
                 offsets.push(total_size);
-                if offsets.windows(2).any(|i| i[0] > i[1]) {
```

Comment:
I don't think this is necessary.
- It's not a performance issue, it's a choice of algorithms.
  Original code has a obvious boundary: do header check first before this line, then do items checks.
  Maybe it's cheaper to do all cheap checks before do recursive checks.
- Reduce readability.
  (As a person who has some kind of compulsive disorder, maybe it just break symmetry.)
  From the code you can see obvious differences between `dynVec` and `Table`.
  Maybe I should extract this part into a single function.

(This is just a comment, not a request for changes.)

Reply:
@yangby-cryptape I've reverted the change according to your comment, please review again, thanks